Game/Scripts/UITextTrigger.cpp

```diff
@@ -20,41 +30,79 @@ UITextTrigger::~UITextTrigger()
 
 void UITextTrigger::Start()
 {
-	player = App->GetModule<ModulePlayer>()->GetPlayer()->GetComponent<ComponentPlayer>();
 	componentRigidBody = owner->GetComponent<ComponentRigidBody>();
+	input = App->GetModule<ModuleInput>();
+
+	textBoxSize = textBox.size() - 1;
+	displacementControl = textBox[textBoxCurrent]->GetComponent<UIImageDisplacementControl>();
 }
 
 void UITextTrigger::Update(float deltaTime)
 {
+	if (dialogueDone)
+	{
+		return;
+	}
+
+	if(textBoxCurrent < textBoxSize)
+	{
+		if (wasInside && input->GetKey(SDL_SCANCODE_F) == KeyState::DOWN)
+		{
+			NextText();
+			displacementControl = textBox[textBoxCurrent]->GetComponent<UIImageDisplacementControl>();
 
+			textBox[textBoxCurrent]->Enable();
```

Comment:
```suggestion
			textBox[static_cast<size_t>(textBoxCurrent)]->Enable();
```
Avoids raising a warning.